source/ceylon/json/Visitor.ceylon

```diff
@@ -0,0 +1,84 @@
+"Callable interface used when traversing JSON data. 
+ 
+ It is the callers responsiblity to ensure the 
+ methods of this interface are called in a sequence 
+ that corresponds to well-formed JSON. 
+ For example, callers should never generate the calling
+ sequence `onKey()`, `onKey()`."
+by("Tom Bentley")
+shared interface Visitor {
```

Comment:
Just for curiosity. Do we really want to use such general names? I'm afraid, that in user code will be collision in names often, and author will have to invent his own name, which cause worse readability for others.